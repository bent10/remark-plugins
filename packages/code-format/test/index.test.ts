/// <reference types="vitest/globals" />

import { remark } from 'remark'
import remarkCodeFormat from '../src/index.js'

it('should format code blocks using Prettier with default options', async () => {
  const md =
    '\n```js prettier\nfoo(reallyLongArg(), omgSoManyParameters(), IShouldRefactorThis(), isThereSeriouslyAnotherOne());\n```'

  const file = await remark().use(remarkCodeFormat).process(md)

  expect(String(file)).toMatchSnapshot()
})

it('should format code blocks using Prettier with specified parser', async () => {
  const md =
    '\n```ts prettier="{ parser: \'typescript\' }"\n@observer class OrderLine {\n  @observable price: number = 0;\n}\n```'

  const file = await remark().use(remarkCodeFormat).process(md)

  expect(String(file)).toMatchSnapshot()
})

it('should handle unsupported code block languages', async () => {
  const md =
    '\n```unsupported-language prettier\nconst user: { [key: string]: string \n| number | boolean } = {\n name: "John Doe", age: 30 };\n```'

  const file = await remark().use(remarkCodeFormat).process(md)

  expect(String(file)).toMatchSnapshot()
})

it('should handle extended language mappings', async () => {
  const md = '```java prettier\npublic class HelloWorld {}\n```'

  const file = await remark().use(remarkCodeFormat).process(md)

  expect(String(file)).toMatchSnapshot()
})

it('should handle formatting failures and preserve the original code block', async () => {
  // invalid TypeScript code to trigger a formatting failure
  const md = '```ts prettier\nconst x number = 1 // missing :\n```'

  const file = await remark().use(remarkCodeFormat).process(md)

  expect(String(file)).toMatchSnapshot()
})
