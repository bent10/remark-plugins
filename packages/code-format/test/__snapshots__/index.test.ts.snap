// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should format code blocks using Prettier with default options 1`] = `
"\`\`\`js
foo(
  reallyLongArg(),
  omgSoManyParameters(),
  IShouldRefactorThis(),
  isThereSeriouslyAnotherOne()
)

\`\`\`
"
`;

exports[`should format code blocks using Prettier with specified parser 1`] = `
"\`\`\`ts
@observer
class OrderLine {
  @observable price: number = 0
}

\`\`\`
"
`;

exports[`should handle extended language mappings 1`] = `
"\`\`\`java
public class HelloWorld {}
\`\`\`
"
`;

exports[`should handle formatting failures and preserve the original code block 1`] = `
"\`\`\`ts
const x number = 1 // missing :
\`\`\`
"
`;

exports[`should handle unsupported code block languages 1`] = `
"\`\`\`unsupported-language prettier
const user: { [key: string]: string 
| number | boolean } = {
 name: \\"John Doe\\", age: 30 };
\`\`\`
"
`;
